top navigation bar fixed

tyler232 opened the pull request.

weikunwu commented on src/Components/utils/Navigation.tsx:
```diff
@@ -0,0 +1,44 @@
+}
+
+const Navigation: React.FC<NavigationProps> = ({
+    aboutUsHandler,
```
Do we need to keep these handlers as props? Or can we just declare them inside the component?

tyler232 replied:
It does seem redundant but what I  was thinking is we keep it as props for now, in case we might change the handler in the future so we don't need to touch this file for a while and all other pages can use this component.

tyler232 reviewed (commented):
It does seem redundant but what I was thinking is we keep it as props for now, in case we might change the handler in the future so we don't need to touch this file for a while and all other pages can use this component. @weikunwu

weikunwu commented:
@tyler232 Plz check if your changes still works after I fix the merge conflict

tyler232 commented:
Compiled with problems:
×
ERROR in ./src/Components/Progress.tsx
Module build failed (from ./node_modules/babel-loader/lib/index.js):
SyntaxError: /Users/changboliu/Desktop/Networking-Platform/src/Components/Progress.tsx: Unexpected token (28:39)

  26 |                     ProgressHandler,
  27 |                     SettingsHandler} : Props) {
> 28 |                     backToHomeHandler} : Props) {
     |                                        ^
  29 |     return (
  30 |         <div style={{ margin: 0, padding: 0, height: '100vh', position: 'relative' }}>
  31 |             <NavigationBar 
    at constructor (/Users/changboliu/Desktop/Networking-Platform/node_modules/@babel/parser/lib/index.js:353:19)
    at TypeScriptParserMixin.raise (/Users/changboliu/Desktop/Networking-Platform/node_modules/@babel/parser/lib/index.js:3277:19)
    at TypeScriptParserMixin.unexpected (/Users/changboliu/Desktop/Networking-Platform/node_modules/@babel/parser/lib/index.js:3297:16)
    at TypeScriptParserMixin.parseExprAtom (/Users/changboliu/Desktop/Networking-Platform/node_modules/@babel/parser/lib/index.js:10964:16)
    at TypeScriptParserMixin.parseExprAtom (/Users/changboliu/Desktop/Networking-Platform/node_modules/@babel/parser/lib/index.js:6821:20)
    at TypeScriptParserMixin.parseExprSubscripts (/Users/changboliu/Desktop/Networking-Platform/node_modules/@babel/parser/lib/index.js:10580:23)
    at TypeScriptParserMixin.parseUpdate (/Users/changboliu/Desktop/Networking-Platform/node_modules/@babel/parser/lib/index.js:10563:21)
    at TypeScriptParserMixin.parseMaybeUnary (/Users/changboliu/Desktop/Networking-Platform/node_modules/@babel/parser/lib/index.js:10541:23)
    at TypeScriptParserMixin.parseMaybeUnary (/Users/changboliu/Desktop/Networking-Platform/node_modules/@babel/parser/lib/index.js:9456:18)
    at TypeScriptParserMixin.parseMaybeUnaryOrPrivate (/Users/changboliu/Desktop/Networking-Platform/node_modules/@babel/parser/lib/index.js:10395:61)
    at TypeScriptParserMixin.parseExprOps (/Users/changboliu/Desktop/Networking-Platform/node_modules/@babel/parser/lib/index.js:10400:23)
    at TypeScriptParserMixin.parseMaybeConditional (/Users/changboliu/Desktop/Networking-Platform/node_modules/@babel/parser/lib/index.js:10377:23)
    at TypeScriptParserMixin.parseMaybeAssign (/Users/changboliu/Desktop/Networking-Platform/node_modules/@babel/parser/lib/index.js:10338:21)
    at TypeScriptParserMixin.parseMaybeAssign (/Users/changboliu/Desktop/Networking-Platform/node_modules/@babel/parser/lib/index.js:9405:20)
    at TypeScriptParserMixin.parseExpressionBase (/Users/changboliu/Desktop/Networking-Platform/node_modules/@babel/parser/lib/index.js:10292:23)
    at /Users/changboliu/Desktop/Networking-Platform/node_modules/@babel/parser/lib/index.js:10288:39
    at TypeScriptParserMixin.allowInAnd (/Users/changboliu/Desktop/Networking-Platform/node_modules/@babel/parser/lib/index.js:11921:16)
    at TypeScriptParserMixin.parseExpression (/Users/changboliu/Desktop/Networking-Platform/node_modules/@babel/parser/lib/index.js:10288:17)
    at TypeScriptParserMixin.parseStatementContent (/Users/changboliu/Desktop/Networking-Platform/node_modules/@babel/parser/lib/index.js:12362:23)
    at TypeScriptParserMixin.parseStatementContent (/Users/changboliu/Desktop/Networking-Platform/node_modules/@babel/parser/lib/index.js:9132:18)
    at TypeScriptParserMixin.parseStatementLike (/Users/changboliu/Desktop/Networking-Platform/node_modules/@babel/parser/lib/index.js:12229:17)
    at TypeScriptParserMixin.parseModuleItem (/Users/changboliu/Desktop/Networking-Platform/node_modules/@babel/parser/lib/index.js:12206:17)
    at TypeScriptParserMixin.parseBlockOrModuleBlockBody (/Users/changboliu/Desktop/Networking-Platform/node_modules/@babel/parser/lib/index.js:12786:36)
    at TypeScriptParserMixin.parseBlockBody (/Users/changboliu/Desktop/Networking-Platform/node_modules/@babel/parser/lib/index.js:12779:10)
    at TypeScriptParserMixin.parseProgram (/Users/changboliu/Desktop/Networking-Platform/node_modules/@babel/parser/lib/index.js:12106:10)
    at TypeScriptParserMixin.parseTopLevel (/Users/changboliu/Desktop/Networking-Platform/node_modules/@babel/parser/lib/index.js:12096:25)
    at TypeScriptParserMixin.parse (/Users/changboliu/Desktop/Networking-Platform/node_modules/@babel/parser/lib/index.js:13895:10)
    at TypeScriptParserMixin.parse (/Users/changboliu/Desktop/Networking-Platform/node_modules/@babel/parser/lib/index.js:9733:18)
    at parse (/Users/changboliu/Desktop/Networking-Platform/node_modules/@babel/parser/lib/index.js:13937:38)
    at parser (/Users/changboliu/Desktop/Networking-Platform/node_modules/@babel/core/lib/parser/index.js:41:34)
    at parser.next (<anonymous>)
    at normalizeFile (/Users/changboliu/Desktop/Networking-Platform/node_modules/@babel/core/lib/transformation/normalize-file.js:64:37)
    at normalizeFile.next (<anonymous>)
    at run (/Users/changboliu/Desktop/Networking-Platform/node_modules/@babel/core/lib/transformation/index.js:21:50)
    at run.next (<anonymous>)
    at transform (/Users/changboliu/Desktop/Networking-Platform/node_modules/@babel/core/lib/transform.js:22:33)
    at transform.next (<anonymous>)
    at step (/Users/changboliu/Desktop/Networking-Platform/node_modules/gensync/index.js:261:32)
    at /Users/changboliu/Desktop/Networking-Platform/node_modules/gensync/index.js:273:13
    at async.call.result.err.err (/Users/changboliu/Desktop/Networking-Platform/node_modules/gensync/index.js:223:11)
ERROR in ./src/Components/utils/BlueNavBar.tsx 12:0-77
Module not found: Error: Can't resolve '../../Images/learning_selected_icon.png' in '/Users/changboliu/Desktop/Networking-Platform/src/Components/utils'
ERROR in ./src/Components/utils/BlueNavBar.tsx 14:0-75
Module not found: Error: Can't resolve '../../Images/setting_selected_icon.png' in '/Users/changboliu/Desktop/Networking-Platform/src/Components/utils'
ERROR
[eslint] 
src/Components/Progress.tsx
  Line 28:39:  Parsing error: Declaration or statement expected

ERROR in ./src/Components/Progress.tsx
Module build failed (from ./node_modules/babel-loader/lib/index.js):
SyntaxError: /Users/changboliu/Desktop/Networking-Platform/src/Components/Progress.tsx: Unexpected token (28:39)

  26 |                     ProgressHandler,
  27 |                     SettingsHandler} : Props) {
> 28 |                     backToHomeHandler} : Props) {
     |                                        ^
  29 |     return (
  30 |         <div style={{ margin: 0, padding: 0, height: '100vh', position: 'relative' }}>
  31 |             <NavigationBar 
    at constructor (/Users/changboliu/Desktop/Networking-Platform/node_modules/@babel/parser/lib/index.js:353:19)
    at TypeScriptParserMixin.raise (/Users/changboliu/Desktop/Networking-Platform/node_modules/@babel/parser/lib/index.js:3277:19)
    at TypeScriptParserMixin.unexpected (/Users/changboliu/Desktop/Networking-Platform/node_modules/@babel/parser/lib/index.js:3297:16)
    at TypeScriptParserMixin.parseExprAtom (/Users/changboliu/Desktop/Networking-Platform/node_modules/@babel/parser/lib/index.js:10964:16)
    at TypeScriptParserMixin.parseExprAtom (/Users/changboliu/Desktop/Networking-Platform/node_modules/@babel/parser/lib/index.js:6821:20)
    at TypeScriptParserMixin.parseExprSubscripts (/Users/changboliu/Desktop/Networking-Platform/node_modules/@babel/parser/lib/index.js:10580:23)
    at TypeScriptParserMixin.parseUpdate (/Users/changboliu/Desktop/Networking-Platform/node_modules/@babel/parser/lib/index.js:10563:21)
    at TypeScriptParserMixin.parseMaybeUnary (/Users/changboliu/Desktop/Networking-Platform/node_modules/@babel/parser/lib/index.js:10541:23)
    at TypeScriptParserMixin.parseMaybeUnary (/Users/changboliu/Desktop/Networking-Platform/node_modules/@babel/parser/lib/index.js:9456:18)
    at TypeScriptParserMixin.parseMaybeUnaryOrPrivate (/Users/changboliu/Desktop/Networking-Platform/node_modules/@babel/parser/lib/index.js:10395:61)
    at TypeScriptParserMixin.parseExprOps (/Users/changboliu/Desktop/Networking-Platform/node_modules/@babel/parser/lib/index.js:10400:23)
    at TypeScriptParserMixin.parseMaybeConditional (/Users/changboliu/Desktop/Networking-Platform/node_modules/@babel/parser/lib/index.js:10377:23)
    at TypeScriptParserMixin.parseMaybeAssign (/Users/changboliu/Desktop/Networking-Platform/node_modules/@babel/parser/lib/index.js:10338:21)
    at TypeScriptParserMixin.parseMaybeAssign (/Users/changboliu/Desktop/Networking-Platform/node_modules/@babel/parser/lib/index.js:9405:20)
    at TypeScriptParserMixin.parseExpressionBase (/Users/changboliu/Desktop/Networking-Platform/node_modules/@babel/parser/lib/index.js:10292:23)
    at /Users/changboliu/Desktop/Networking-Platform/node_modules/@babel/parser/lib/index.js:10288:39
    at TypeScriptParserMixin.allowInAnd (/Users/changboliu/Desktop/Networking-Platform/node_modules/@babel/parser/lib/index.js:11921:16)
    at TypeScriptParserMixin.parseExpression (/Users/changboliu/Desktop/Networking-Platform/node_modules/@babel/parser/lib/index.js:10288:17)
    at TypeScriptParserMixin.parseStatementContent (/Users/changboliu/Desktop/Networking-Platform/node_modules/@babel/parser/lib/index.js:12362:23)
    at TypeScriptParserMixin.parseStatementContent (/Users/changboliu/Desktop/Networking-Platform/node_modules/@babel/parser/lib/index.js:9132:18)
    at TypeScriptParserMixin.parseStatementLike (/Users/changboliu/Desktop/Networking-Platform/node_modules/@babel/parser/lib/index.js:12229:17)
    at TypeScriptParserMixin.parseModuleItem (/Users/changboliu/Desktop/Networking-Platform/node_modules/@babel/parser/lib/index.js:12206:17)
    at TypeScriptParserMixin.parseBlockOrModuleBlockBody (/Users/changboliu/Desktop/Networking-Platform/node_modules/@babel/parser/lib/index.js:12786:36)
    at TypeScriptParserMixin.parseBlockBody (/Users/changboliu/Desktop/Networking-Platform/node_modules/@babel/parser/lib/index.js:12779:10)
    at TypeScriptParserMixin.parseProgram (/Users/changboliu/Desktop/Networking-Platform/node_modules/@babel/parser/lib/index.js:12106:10)
    at TypeScriptParserMixin.parseTopLevel (/Users/changboliu/Desktop/Networking-Platform/node_modules/@babel/parser/lib/index.js:12096:25)
    at TypeScriptParserMixin.parse (/Users/changboliu/Desktop/Networking-Platform/node_modules/@babel/parser/lib/index.js:13895:10)
    at TypeScriptParserMixin.parse (/Users/changboliu/Desktop/Networking-Platform/node_modules/@babel/parser/lib/index.js:9733:18)
    at parse (/Users/changboliu/Desktop/Networking-Platform/node_modules/@babel/parser/lib/index.js:13937:38)
    at parser (/Users/changboliu/Desktop/Networking-Platform/node_modules/@babel/core/lib/parser/index.js:41:34)
    at parser.next (<anonymous>)
    at normalizeFile (/Users/changboliu/Desktop/Networking-Platform/node_modules/@babel/core/lib/transformation/normalize-file.js:64:37)
    at normalizeFile.next (<anonymous>)
    at run (/Users/changboliu/Desktop/Networking-Platform/node_modules/@babel/core/lib/transformation/index.js:21:50)
    at run.next (<anonymous>)
    at transform (/Users/changboliu/Desktop/Networking-Platform/node_modules/@babel/core/lib/transform.js:22:33)
    at transform.next (<anonymous>)
    at step (/Users/changboliu/Desktop/Networking-Platform/node_modules/gensync/index.js:261:32)
    at /Users/changboliu/Desktop/Networking-Platform/node_modules/gensync/index.js:273:13
    at async.call.result.err.err (/Users/changboliu/Desktop/Networking-Platform/node_modules/gensync/index.js:223:11)
ERROR in ./src/Components/utils/BlueNavBar.tsx 12:0-77
Module not found: Error: Can't resolve '../../Images/learning_selected_icon.png' in '/Users/changboliu/Desktop/Networking-Platform/src/Components/utils'
ERROR in ./src/Components/utils/BlueNavBar.tsx 14:0-75
Module not found: Error: Can't resolve '../../Images/setting_selected_icon.png' in '/Users/changboliu/Desktop/Networking-Platform/src/Components/utils'
ERROR
[eslint] 
src/Components/Progress.tsx
  Line 28:39:  Parsing error: Declaration or statement expected

ERROR in src/App.tsx:90:42
TS2741: Property 'SettingsHandler' is missing in type '{ ProfileHandler: () => void; aboutUsHandler: () => void; contactHandler: () => void; FAQHandler: () => void; getStartedHandler: () => void; InventoryHandler: () => void; backToHomeHandler: () => void; ProgressHandler: () => void; }' but required in type 'Props'.
    88 |                                         FAQHandler={toFAQPage} 
    89 |                                       />} />
  > 90 |         <Route path="/profile" element={<Profile 
       |                                          ^^^^^^^
    91 |                                             ProfileHandler={toProfilePage} 
    92 |                                             aboutUsHandler={toAboutUsPage} 
    93 |                                             contactHandler={toContactPage} 
ERROR in src/App.tsx:100:44
TS2741: Property 'SettingsHandler' is missing in type '{ ProfileHandler: () => void; aboutUsHandler: () => void; contactHandler: () => void; FAQHandler: () => void; getStartedHandler: () => void; ViewHandler: () => void; InventoryHandler: () => void; backToHomeHandler: () => void; ProgressHandler: () => void; }' but required in type 'Props'.
     98 |                                             ProgressHandler={toProgressPage}
     99 |                                           />} />
  > 100 |         <Route path="/inventory" element={<Inventory 
        |                                            ^^^^^^^^^
    101 |                                               ProfileHandler={toProfilePage} 
    102 |                                               aboutUsHandler={toAboutUsPage} 
    103 |                                               contactHandler={toContactPage} 
ERROR in src/App.tsx:111:49
TS2739: Type '{ ProfileHandler: () => void; aboutUsHandler: () => void; contactHandler: () => void; FAQHandler: () => void; getStartedHandler: () => void; ViewHandler: () => void; InventoryHandler: () => void; backToHomeHandler: () => void; }' is missing the following properties from type 'Props': SettingsHandler, ProgressHandler
    109 |                                               ProgressHandler={toProgressPage}
    110 |                                             />} />
  > 111 |         <Route path="/inventory/view" element={<View 
        |                                                 ^^^^
    112 |                                               ProfileHandler={toProfilePage} 
    113 |                                               aboutUsHandler={toAboutUsPage} 
    114 |                                               contactHandler={toContactPage} 
ERROR in src/App.tsx:121:43
TS2741: Property 'SettingsHandler' is missing in type '{ ProfileHandler: () => void; aboutUsHandler: () => void; contactHandler: () => void; FAQHandler: () => void; getStartedHandler: () => void; InventoryHandler: () => void; ProgressHandler: () => void; backToHomeHandler: () => void; }' but required in type 'Props'.
    119 |                                               backToHomeHandler={backToHome}
    120 |                                             />} />
  > 121 |         <Route path="/progress" element={<Progress 
        |                                           ^^^^^^^^
    122 |                                               ProfileHandler={toProfilePage} 
    123 |                                               aboutUsHandler={toAboutUsPage} 
    124 |                                               contactHandler={toContactPage} 
ERROR in src/App.tsx:121:43
TS2786: 'Progress' cannot be used as a JSX component.
  Its return type 'void' is not a valid JSX element.
    119 |                                               backToHomeHandler={backToHome}
    120 |                                             />} />
  > 121 |         <Route path="/progress" element={<Progress 
        |                                           ^^^^^^^^
    122 |                                               ProfileHandler={toProfilePage} 
    123 |                                               aboutUsHandler={toAboutUsPage} 
    124 |                                               contactHandler={toContactPage} 
ERROR in src/Components/Progress.tsx:28:21
TS2304: Cannot find name 'backToHomeHandler'.
    26 |                     ProgressHandler,
    27 |                     SettingsHandler} : Props) {
  > 28 |                     backToHomeHandler} : Props) {
       |                     ^^^^^^^^^^^^^^^^^
    29 |     return (
    30 |         <div style={{ margin: 0, padding: 0, height: '100vh', position: 'relative' }}>
    31 |             <NavigationBar 
ERROR in src/Components/Progress.tsx:28:40
TS1128: Declaration or statement expected.
    26 |                     ProgressHandler,
    27 |                     SettingsHandler} : Props) {
  > 28 |                     backToHomeHandler} : Props) {
       |                                        ^
    29 |     return (
    30 |         <div style={{ margin: 0, padding: 0, height: '100vh', position: 'relative' }}>
    31 |             <NavigationBar 
ERROR in src/Components/Progress.tsx:28:42
TS1434: Unexpected keyword or identifier.
    26 |                     ProgressHandler,
    27 |                     SettingsHandler} : Props) {
  > 28 |                     backToHomeHandler} : Props) {
       |                                          ^^^^^
    29 |     return (
    30 |         <div style={{ margin: 0, padding: 0, height: '100vh', position: 'relative' }}>
    31 |             <NavigationBar 
ERROR in src/Components/Progress.tsx:28:42
TS2693: 'Props' only refers to a type, but is being used as a value here.
    26 |                     ProgressHandler,
    27 |                     SettingsHandler} : Props) {
  > 28 |                     backToHomeHandler} : Props) {
       |                                          ^^^^^
    29 |     return (
    30 |         <div style={{ margin: 0, padding: 0, height: '100vh', position: 'relative' }}>
    31 |             <NavigationBar 
ERROR in src/Components/Progress.tsx:28:47
TS1128: Declaration or statement expected.
    26 |                     ProgressHandler,
    27 |                     SettingsHandler} : Props) {
  > 28 |                     backToHomeHandler} : Props) {
       |                                               ^
    29 |     return (
    30 |         <div style={{ margin: 0, padding: 0, height: '100vh', position: 'relative' }}>
    31 |             <NavigationBar 
    
    I'll try to fix it to get it work

tyler232 commented:
@weikunwu I fixed majority of the error, but for those 2 errors in BlueNavBar.tsx below I'm not able to fix. There are no learning_selected_icon.png and setting_selected_icon.png locally. 

```
ERROR in ./src/Components/utils/BlueNavBar.tsx 12:0-77
Module not found: Error: Can't resolve '../../Images/learning_selected_icon.png' in '/Users/changboliu/Desktop/Networking-Platform/src/Components/utils'

ERROR in ./src/Components/utils/BlueNavBar.tsx 14:0-75
Module not found: Error: Can't resolve '../../Images/setting_selected_icon.png' in '/Users/changboliu/Desktop/Networking-Platform/src/Components/utils'
```

I've pushed my change already:
fixed some syntax issues in Progress.tsx 
Some people edited Props for Inventory, Profile, Views, Progress. But App.tsx still wasn't updated, therefore causing errors, I've changed to the correct props input in App.tsx
The last thing is for BlueNavBar's image, those 2 images don't exist locally so I'm not able to fix it.

tyler232 commented:
Moved to the new PR
